### PR TITLE
Remove configuration page and migrate to MetaBrainz.MusicBrainz.CoverArt

### PR DESCRIPTION
**Pull request by heyhippari**
* Removes the placeholder configuration page
* Move to [MetaBrainz.MusicBrainz.CoverArt](https://github.com/Zastai/MetaBrainz.MusicBrainz.CoverArt), the MetaBrainz-sanctionned library for the Cover Art Archive

### CODE REVIEW

File: Jellyfin.Plugin.CoverArtArchive/CoverArtArchiveImageProvider.cs [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+namespace Jellyfin.Plugin.CoverArtArchive
+{
+    /// <summary>
+    /// The cover art archive image provider.
+    /// </summary>
+    public class CoverArtArchiveImageProvider : IRemoteImageProvider
+    {
+        private readonly ILogger<CoverArtArchiveImageProvider> _logger;
+        private readonly IHttpClientFactory _httpClientFactory;
+        private readonly JsonSerializerOptions _serializerOptions;
```
Comment (crobibero) [inline Jellyfin.Plugin.CoverArtArchive/CoverArtArchiveImageProvider.cs]:
```suggestion
```

This is never used

(also clean up constructor and `using` declarations)

File: Jellyfin.Plugin.CoverArtArchive/CoverArtArchiveImageProvider.cs [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+                    var frontImage = release.Images.SingleOrDefault(image => image.Types == CoverArtType.Front);
+
+                    if (frontImage is not null)
+                    {
+                        list.Add(new RemoteImageInfo
+                        {
+                            ProviderName = Name,
+                            Url = frontImage.Location!.ToString(),
+                            Type = ImageType.Primary,
+                            ThumbnailUrl = ((frontImage.Thumbnails.Large ?? frontImage.Thumbnails.Small)!).ToString(),
```
Comment (crobibero) [inline Jellyfin.Plugin.CoverArtArchive/CoverArtArchiveImageProvider.cs]:
```suggestion
                            ThumbnailUrl = (frontImage.Thumbnails.Large ?? frontImage.Thumbnails.Small)?.ToString(),
```

File: Jellyfin.Plugin.CoverArtArchive/CoverArtArchiveImageProvider.cs [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+                        var frontImage = release.Images.SingleOrDefault(image => image.Types == CoverArtType.Front);
+
+                        if (frontImage is not null)
+                        {
+                            list.Add(new RemoteImageInfo
+                            {
+                                ProviderName = Name,
+                                Url = frontImage.Location!.ToString(),
+                                Type = ImageType.Primary,
+                                ThumbnailUrl = ((frontImage.Thumbnails.Large ?? frontImage.Thumbnails.Small)!).ToString(),
```
Comment (crobibero) [inline Jellyfin.Plugin.CoverArtArchive/CoverArtArchiveImageProvider.cs]:
```suggestion
                                ThumbnailUrl = (frontImage.Thumbnails.Large ?? frontImage.Thumbnails.Small)?.ToString(),
```

File: Jellyfin.Plugin.CoverArtArchive/CoverArtArchiveImageProvider.cs [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+            var list = new List<RemoteImageInfo>();
+            var musicBrainzId = item.GetProviderId(MetadataProvider.MusicBrainzAlbum);
+
+            if (!string.IsNullOrEmpty(musicBrainzId))
+            {
+                try
+                {
+                    var release = await _coverArtClient.FetchReleaseAsync(Guid.Parse(musicBrainzId)).ConfigureAwait(false);
+
+                    var frontImage = release.Images.SingleOrDefault(image => image.Types == CoverArtType.Front);
```
Comment (crobibero) [inline Jellyfin.Plugin.CoverArtArchive/CoverArtArchiveImageProvider.cs]:
```suggestion
                    var frontImage = release.Images.FirstOrDefault(image => image.Types == CoverArtType.Front);
```

Just a reminder that `SingleOrDefault` will throw an exception if multiple matches are found, and it fully iterates through the list.

File: Jellyfin.Plugin.CoverArtArchive/CoverArtArchiveImageProvider.cs [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+            if (list.Count == 0)
+            {
+                var musicBrainzGroupId = item.GetProviderId(MetadataProvider.MusicBrainzReleaseGroup);
+                if (!string.IsNullOrEmpty(musicBrainzGroupId))
+                {
+                    try
+                    {
+                        var release = await _coverArtClient.FetchGroupReleaseAsync(Guid.Parse(musicBrainzGroupId)).ConfigureAwait(false);
+
+                        var frontImage = release.Images.SingleOrDefault(image => image.Types == CoverArtType.Front);
```
Comment (crobibero) [inline Jellyfin.Plugin.CoverArtArchive/CoverArtArchiveImageProvider.cs]:
```suggestion
                        var frontImage = release.Images.FirstOrDefault(image => image.Types == CoverArtType.Front);
```
Same

**Review by crobibero — APPROVED** [review]

**Comment by crobibero** [comment]
Blocked due to upstream license - https://github.com/Zastai/MetaBrainz.MusicBrainz.CoverArt/issues/17

**Comment by ghost** [comment]
> Blocked due to upstream license - [Zastai/MetaBrainz.MusicBrainz.CoverArt#17](https://github.com/Zastai/MetaBrainz.MusicBrainz.CoverArt/issues/17)

FYI, license has been updated for MetaBrainz.MusicBrainz.CoverArt v4.0.0.

**Pull request merged**